snykTask/src/task-args.ts

```diff
@@ -150,6 +155,18 @@ class TaskArgs {
       const errorMsg = `If set, codeSeverityThreshold must be one from [${taskTestTypeThreshold}] (case insensitive). If not set, the default is '${Severity.LOW}'.`;
       throw new Error(errorMsg);
     }
+
+    // code output json dependent on tested severity threshold.
+    if (
+      taskTestType === TestType.CODE &&
+      ((this.codeSeverityThreshold &&
+        getSeverityOrdinal(this.failOnThreshold) <
+          getSeverityOrdinal(this.codeSeverityThreshold)) ||
+        (!this.codeSeverityThreshold && this.failOnThreshold !== Severity.LOW))
+    ) {
+      const errorMsg = `If set, failOnThreshold must be matching or higher severity than its codeSeverityThreshold.`;
+      throw new Error(errorMsg);
+    }
```

Comment:
question: why is this logic there?

Reply:
This validation check is now omitted, consistent with sca scans. It is still semantically correct to run a test (sca, sast) with `severityThreshold` or `codeSeverityThreshold` on a higher severity than its `failOnThreshold`. It means output json results returned with issues will satisfy `failOnThreshold` value. In that sense, `failOnThreshold` value is redundant so does not need to be validated.